### PR TITLE
feat: re-export Promise to get better typing.

### PR DESCRIPTION
**Pull request by mprobst**
[Review on Reviewable](https://reviewable.io/reviews/angular/angular/2715)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -123,7 +123,7 @@
     "through2": "^0.6.1",
     "ts2dart": "^0.6.0",
     "tsd": "^0.6.0",
-    "typescript": "alexeagle/TypeScript#error_is_class",
+    "typescript": "microsoft/TypeScript#release-1.5",
```
Comment (rkirov) [inline package.json]: do you need to regen shrinkwrap?

**Comment by mprobst** [comment]
Sorry, this is not ready for review, I'm waiting on a fix from TypeScript.

**Comment by mprobst** [comment]
This is ready for review, but will be pending on TypeScript making progresson https://github.com/Microsoft/TypeScript/issues/3640.

**Comment by tbosch** [comment]
@mprobst Is this still blocked as the Typescript issue is closed?

**Comment by mprobst** [comment]
@alexeagle could you TAL? @rkirov is OOO.

**Comment by mprobst** [comment]
Merged as a62a6ba40a86272021979fe955f1d4970a6b195b.

**Pull request closed without merging**